xdsl/builder.py

```diff
@@ -114,6 +114,9 @@ def region(
         else:
             return Builder._region_args(input)
 
+    def implicit(self) -> ImplicitBuilder:
```

Comment:
Can you add here a reference to `ImplicitBuilder` as documentation?

Reply:
I'm a bit confused, it's a command/control click away

Reply:
Well yes, but at this point, especially for new people, what people see is just a function creating something else. They don't necessarily know how to use it, and having just a bit of documentation can help.
Just something like "Create an implicit builder that is used in a `with` region. See `ImplicitBuilder`" can really help people understanding what this is supposed to be. Otherwise, people might just skip over it?
What do you think?

Reply:
Wouldn't your first instinct be to look at what the return type is? I doubt that many people will skip over it, although maybe I'm wrong. In VSCode you wouldn't even need to click on it, hovering over the type will bring up the docstring

Reply:
Well, yeah when you are looking at the operation.
But when you are looking at a use of `implicit`, you would like to know what it does (meaning, that it creates a new implicit context). Here, you would have to jump to the `implicit` definition to understand what it does, and then to jump to the `ImplicitBuilder`.
That's why I would say that a single sentence would still help, but if you think I'm too hard on that, we could skip it.

Reply:
No I think you're right, but in some ways doccomments are a liability, in that there's no way to test that they don't become outdated as we change the code. I'd like to keep documentation as close as possible to the place where the code is defined, so that it's easier to update things as they evolve, and avoid misleading people. I think we can assume that the kind of people who read documentation can navigate to the definition of a class they are interested in. Especially since I don't expect that this function will be used explicitly. Now that I think about it, maybe I should just delete this function and use `ImplicitBuilder(block)` instead... I'll try it out.